src/hooks/useFetch.ts

```diff
@@ -28,10 +30,17 @@ export const useFetch = <T>(fetcher: () => Promise<T>, deps: unknown[] = []) =>
         if (isMounted) {
           setFetchState({ isLoading: false, isError: false, data })
         }
-      } catch (error) {
+      } catch (error: unknown) {
         console.error('Fetch error:', error)
+
+        // ! 특정 에러 상태 처리를 위해 추가
+        let errorStatus: number | undefined = undefined
+        if (typeof error === 'object' && error !== null && 'response' in error) {
+          const err = error as AxiosError
+          errorStatus = err.response?.status
+        }
```

Comment:
instanceof를 활용해 조건문을 간결하게 할 수 있어보여요 😊

```suggestion
        if (error instanceof AxiosError && error.response) {
          const errorStatus = error.response.status
          setFetchState({ isLoading: false, isError: true, data: null, errorStatus })
        }
```